src/js/angular/import/directives/import-resource-tree.directive.js

```diff
@@ -210,83 +219,11 @@ function importResourceTreeDirective($timeout, ImportContextService) {
-
-            const compareByName = (acs) => (r1, r2) => {
-                return acs ? r1.importResource.name.localeCompare(r2.importResource.name) : r2.importResource.name.localeCompare(r1.importResource.name);
-            };
-
-            const compareBySize = (acs) => (r1, r2) => {
-                // The format of size returned by the backend has changed, but we need to keep the old format for backward compatibility.
-                // Therefore, we convert the size to always be in bytes.
-                const r1Size = convertToBytes(r1.importResource.size);
-                const r2Size = convertToBytes(r2.importResource.size);
-                return acs ? r1Size - r2Size : r2Size - r1Size;
-            };
-
-            const compareByModified = (acs) => (r1, r2) => {
-                const r1ModifiedOn = r1.importResource.modifiedOn || Number.MAX_VALUE;
-                const r2ModifiedOn = r2.importResource.modifiedOn || Number.MAX_VALUE;
-                return acs ? r1ModifiedOn - r2ModifiedOn : r2ModifiedOn - r1ModifiedOn;
-            };
-
-            const compareByImportedOn = (acs) => (r1, r2) => {
-                const r1ImportedOn = r1.importResource.importedOn || Number.MAX_VALUE;
-                const r2ImportedOn = r2.importResource.importedOn || Number.MAX_VALUE;
-                return acs ? r1ImportedOn - r2ImportedOn : r2ImportedOn - r1ImportedOn;
-            };
-
-            const compareByContext = (acs) => (r1, r2) => {
-                return acs ? r1.importResource.context.localeCompare(r2.importResource.context) : r2.importResource.context.localeCompare(r1.importResource.context);
-            };
-
-            const filterByType = (resource) => {
-                if (TYPE_FILTER_OPTIONS.ALL === $scope.filterByType) {
-                    return true;
-                }
-
-                if ($scope.filterByType === TYPE_FILTER_OPTIONS.FILE) {
-                    return resource.isFile();
-                }
-
-                if ($scope.filterByType === TYPE_FILTER_OPTIONS.DIRECTORY) {
-                    return resource.isDirectory();
-                }
-
-                return false;
-            };
-
-            const filterByName = (resource) => {
-                if (!$scope.filterByFileName) {
-                    return true;
-                }
-
-                if ($scope.filterByType === TYPE_FILTER_OPTIONS.DIRECTORY) {
-                    return resource.hasTextInDirectoriesName($scope.filterByFileName);
-                }
+                const mainCheckbox = element[0].querySelector('#importSelectCheckboxInput');
 
-                if ($scope.filterByType === TYPE_FILTER_OPTIONS.FILE) {
-                    return resource.hasTextInFilesName($scope.filterByFileName);
+                if (mainCheckbox) {
+                    mainCheckbox.checked = !!($scope.resourceListUtil.areAllDisplayedImportResourcesSelected || $scope.resourceListUtil.areAllDisplayedImportResourcesPartialSelected);
```

Comment:
This condition `!!($scope.resourceListUtil.areAllDisplayedImportResourcesSelected || $scope.resourceListUtil.areAllDisplayedImportResourcesPartialSelected) ` could be moved into a function in `CheckboxControlModel`